vm-bootstrap.sh

```diff
@@ -5,30 +5,116 @@ set -e
 
 # prepare the machine
 sudo apt-get update
-sudo apt-get -y install git autoconf build-essential gperf bison flex texinfo libtool libncurses5-dev wget gawk libc6-dev-amd64 unzip
-sudo mkdir /opt/Espressif || :
+sudo apt-get -y install git autoconf build-essential \
+     gperf bison flex texinfo libtool libncurses5-dev \
+     wget gawk libc6-dev-amd64 python-serial libexpat-dev unzip
+if [ ! -d /opt/Espressif ]; then
+	sudo mkdir /opt/Espressif
+fi
 sudo chown vagrant /opt/Espressif
 
 # Build the cross-compiler
-cp -r /vagrant/tools/crosstool-NG /opt/Espressif/
+cd /opt/Espressif
+if [ ! -d /opt/Espressif/crosstool-NG ]; then
+	git clone -b lx106 git://github.com/jcmvbkbc/crosstool-NG.git
+fi
+
 cd /opt/Espressif/crosstool-NG
-./bootstrap && ./configure -- prefix=`pwd` && make && make install
-./ct-ng xtensa-lx106-elf
-./ct-ng build
+git pull
+
+if [ ! -d /opt/Espressif/crosstool-NG/builds ]; then
+	./bootstrap && ./configure -- prefix=`pwd` && make && make install
+	./ct-ng xtensa-lx106-elf
+fi
+
+
+CT_DEBUG_CT_SAVE_STEPS=1 ./ct-ng build
+PATH=$PWD/builds/xtensa-lx106-elf/bin:$PATH
+
+# Setup the cross compiler
+HAS_PATH=`cat ~/.bashrc | grep "/opt/Espressif/crosstool-NG/builds/xtensa-lx106-elf/bin:" || echo -n ""`
```

Comment:
ah, nevermind, need them so the script doesn't halt due to `set -e`.  i guess in this case i prefer the `|| :` idiom